src/tests/src/libs/antares/study/parallel-years/test-parallel-years-calculator.cpp

```diff
@@ -0,0 +1,354 @@
+#define BOOST_TEST_MODULE test-end-to-end tests
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/included/unit_test.hpp>
+
+#include <antares/study/study.h>
+#include <antares/study/parallel-years.h>
+
+namespace utf = boost::unit_test;
+namespace tt = boost::test_tools;
+
+using namespace Yuni;
+using namespace Antares::Data;
+
+struct Fixture
+{
+	Fixture(const Fixture& f) = delete;
+	Fixture(const Fixture&& f) = delete;
+	Fixture& operator= (const Fixture& f) = delete;
+	Fixture& operator= (const Fixture&& f) = delete;
+	Fixture() 
+	{
+        // using default params
+        params.reset();
+	}
+
+	~Fixture(){}
+
+    Parameters params{};
+    bool forceParallel = false;
+    bool enableParallel = false;
+    uint maxNbYearsInParallel = 0;
+	
+};
+
+BOOST_FIXTURE_TEST_SUITE(s, Fixture)
+
+BOOST_AUTO_TEST_CASE(default_params_no_force)
+{
+    bool thermalTSRefresh = false;
+
+    params.nbCores.ncMode = Antares::Data::NumberOfCoresMode::ncHigh;
+
+    SetsOfParallelYearCalculator builder(forceParallel,
+                                         enableParallel,
+                                         maxNbYearsInParallel, 1,
```

Comment:
The number of available cores on the machine is quite hidden : here it's **1**.
I would recommend to declare a new variable `uint machine_nb_cores = 1;` and pass it to the constructor below.